runner_service/utils.py

```diff
@@ -206,18 +207,21 @@ def timeout_handler():
             raise SSHTimeout
 
         socket.setdefaulttimeout(self.timeout)
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         try:
-            s.connect((self.host, self.port))
+            family, type, proto, canonname, sockaddr = socket.getaddrinfo(self.host, self.port, 0, 0, socket.SOL_TCP)[0]
         except socket.gaierror:
             raise HostNotFound
-        except ConnectionRefusedError:
-            raise SSHNotAccessible
-        except socket.timeout:
-            raise SSHTimeout
-        else:
-            s.shutdown(socket.SHUT_RDWR)
-            s.close()
+
+        with socket.socket(family, socket.SOCK_STREAM) as s:
```

Comment:
We were specifying `socket.SOL_TCP` during `socket.getaddrinfo` as protocol, so let's do the same here: `(family, socket.SOCK_STREAM, socket.SOL_TCP)`